internal/oci/ecr/ecr.go

```diff
@@ -0,0 +1,63 @@
+package ecr
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ecr"
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+var ErrNoAWSECRAuthorizationData = errors.New("no ecr authorization data provided")
+
+type Client interface {
+	GetAuthorizationToken(ctx context.Context, params *ecr.GetAuthorizationTokenInput, optFns ...func(*ecr.Options)) (*ecr.GetAuthorizationTokenOutput, error)
+}
+
+type ECR struct {
+	client Client
+}
+
+func (r *ECR) CredentialFunc(registry string) auth.CredentialFunc {
+	cfg, err := config.LoadDefaultConfig(context.Background())
+	if err != nil {
+		return func(context.Context, string) (auth.Credential, error) {
+			return auth.EmptyCredential, err
+		}
+	}
+	r.client = ecr.NewFromConfig(cfg)
```

Comment:
The configuration needs to be loaded everytime the `Credential` method is called! The reason is from my issue I told about `AWS_WEB_IDENTITY_TOKEN_FILE` and one specific thing I didn't mention is: This file contains a token used to make the authentication but it also has an expiration time, EKS will rotate the file content for us!

We just need to load the config everytime!

Reply:
Thank you for this update. I changed the code a bit to address that. 

One thing which worry me if it will work without `region` option.  It would be great to support both `IRSA` and `EKS Pod Identity` without extra configuration.

Reply:
Yeah, that all will work! Fortunately `aws sdk` works well with configs using environment variables! Some sources:
* [Specifying the AWS Region](https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/#specifying-the-aws-region): The region can be configured using the `AWS_REGION` env var or passing to the `config.LoadDefaultConfig`
* [Specifying Credentials](https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/#specifying-credentials): Works well with static credentials and web identity tokens env vars as with credentials files and others

So the user just need to set the env vars correctly, this is already happens when we look at the `S3 Storage` option

Reply:
@thepabloaguilar Could you please give a try in your local environment before PR is merged this into `main`?

Reply:
Yeap, I can test it but just at today's night! But the code seems to do what it's intended to do